static_report/src/components/Common/CommonModal.tsx

```diff
@@ -0,0 +1,38 @@
+import {
+  Modal,
+  ModalProps,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  UseDisclosureReturn,
+} from '@chakra-ui/react';
+import type { ReactNode } from 'react';
+
+interface CommonModalProps extends UseDisclosureReturn, ModalProps {
+  title?: ReactNode | string;
+  children: ReactNode | string;
+  footer?: ReactNode;
+}
+
+export function CommonModal({
```

Comment:
A number of our components use the modal disclosure API directly. Should those be refactored?
e.g. `Sidebar`, `CRAssertionModal`, `DeleteModal` ...etc

Since this is a "Common" modal ;-)

Reply:
Haha, for sure, let me refactor it in the next commit.